### PR TITLE
feat(lambda-trigger): expose Lambda invocation type. Closes: #994

### PR DESCRIPTION
**Pull request by whynowy**
Signed-off-by: Derek Wang <whynowy@gmail.com>

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

File: pkg/apis/sensor/v1alpha1/types.go [hunk 1]
```diff
@@ -435,13 +435,25 @@ type AWSLambdaTrigger struct {
...
+	//    The API response includes the function response and additional data.
+	//
+	//    * Event - Invoke the function asynchronously. Send events that fail multiple
+	//    times to the function's dead-letter queue (if it's configured). The API
+	//    response only includes a status code.
+	//
+	//    * DryRun - Validate parameter values and verify that the user or role
+	//    has permission to invoke the function.
+	// +optional
+	InvocationType *string `json:"invocationType,omitempty" protobuf:"bytes,7,opt,name=invocationType"`
```
Comment (alexec) [inline pkg/apis/sensor/v1alpha1/types.go:456]: in empty string valid? if not, change to non-pointer type

**Review by alexec — APPROVED** [review]

File: pkg/apis/sensor/v1alpha1/types.go (hunk 1 above)
Comment (whynowy) [reply pkg/apis/sensor/v1alpha1/types.go:456]: No, it's not. 

The AWS API uses `*string`, so the most convenient way is to use the same type.

Comment (alexec) [reply pkg/apis/sensor/v1alpha1/types.go:456]: I don't feel strongly.

Comment (whynowy) [reply pkg/apis/sensor/v1alpha1/types.go:456]: Why?

Precisely, we are not sure if empty string is valid or not, that depends on AWS server side implementation. But we do know `nil` is valid as default value.

So if we use `string`, we should do something as below:

```
        input := &lambda.InvokeInput{
		FunctionName:   &trigger.FunctionName,
		Payload:        payload,
	}
        if trigger.InvocationType != "" {
                input.InvocationType = &trigger.InvocationType
        }
```

But if we use `*string`, we only need to do:

```
        input := &lambda.InvokeInput{
		FunctionName:   &trigger.FunctionName,
		Payload:        payload,
		InvocationType: trigger.InvocationType,
	}
```

Comment (alexec) [reply pkg/apis/sensor/v1alpha1/types.go:456]: I read your second comment - and misunderstood - sounds like it needs to be *string

**Pull request merged**